<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodeCountLine</class>
 <widget class="QMainWindow" name="CodeCountLine">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1137</width>
    <height>736</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Lucida Console</family>
    <pointsize>14</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>CodeCountLine</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="6" column="0">
       <layout class="QGridLayout" name="gridLayoutBottom">
        <item row="0" column="2" rowspan="3">
         <layout class="QGridLayout" name="gridLayoutButton">
          <item row="0" column="1">
           <widget class="QPushButton" name="fileCreate">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="palette">
             <palette>
              <active>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </active>
              <inactive>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </inactive>
              <disabled>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>120</red>
                  <green>120</green>
                  <blue>120</blue>
                 </color>
                </brush>
               </colorrole>
              </disabled>
             </palette>
            </property>
            <property name="font">
             <font>
              <family>Lucida Console</family>
              <pointsize>16</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Создать
файл с кодом</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="start">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="palette">
             <palette>
              <active>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </active>
              <inactive>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </inactive>
              <disabled>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>120</red>
                  <green>120</green>
                  <blue>120</blue>
                 </color>
                </brush>
               </colorrole>
              </disabled>
             </palette>
            </property>
            <property name="font">
             <font>
              <family>Lucida Console</family>
              <pointsize>16</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Считать</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="copyBuffer">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="palette">
             <palette>
              <active>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </active>
              <inactive>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </inactive>
              <disabled>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>120</red>
                  <green>120</green>
                  <blue>120</blue>
                 </color>
                </brush>
               </colorrole>
              </disabled>
             </palette>
            </property>
            <property name="font">
             <font>
              <family>Lucida Console</family>
              <pointsize>16</pointsize>
             </font>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="text">
             <string>Копировать
таблицу</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="fileConfig">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="palette">
             <palette>
              <active>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </active>
              <inactive>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>174</red>
                  <green>10</green>
                  <blue>179</blue>
                 </color>
                </brush>
               </colorrole>
              </inactive>
              <disabled>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>120</red>
                  <green>120</green>
                  <blue>120</blue>
                 </color>
                </brush>
               </colorrole>
              </disabled>
             </palette>
            </property>
            <property name="font">
             <font>
              <family>Lucida Console</family>
              <pointsize>16</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Настройки
файла</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="resLineLabel">
          <property name="text">
           <string>Количество строк кода:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="resFile">
          <property name="palette">
           <palette>
            <active>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>120</red>
                <green>120</green>
                <blue>120</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
          <property name="font">
           <font>
            <family>Lucida Console</family>
            <pointsize>22</pointsize>
           </font>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="resFileLabel">
          <property name="text">
           <string>Количество файлов:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="resTime">
          <property name="palette">
           <palette>
            <active>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>120</red>
                <green>120</green>
                <blue>120</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
          <property name="font">
           <font>
            <family>Lucida Console</family>
            <pointsize>22</pointsize>
           </font>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="resLine">
          <property name="palette">
           <palette>
            <active>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>31</red>
                <green>11</green>
                <blue>188</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>120</red>
                <green>120</green>
                <blue>120</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
          <property name="font">
           <font>
            <family>Lucida Console</family>
            <pointsize>22</pointsize>
           </font>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="resTimeLabel">
          <property name="text">
           <string>Время работы</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <layout class="QHBoxLayout" name="horizontalLayoutTop2">
        <item>
         <widget class="QCheckBox" name="calculeteSpaceLine">
          <property name="text">
           <string>Учитывать
пустые строки</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="pathLabel">
          <property name="text">
           <string>Путь к папке с кодом</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="path"/>
        </item>
        <item>
         <widget class="QPushButton" name="getPath">
          <property name="text">
           <string>Выбрать папку</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="5" column="0">
       <layout class="QGridLayout" name="gridLayoutCentral">
        <item row="0" column="0">
         <widget class="QTableWidget" name="tableWidget">
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Имя файла</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Размер файла в байтах</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Дата создания</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Контрольная сумма</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayoutTop">
        <item row="0" column="1">
         <widget class="QLineEdit" name="templateSearch">
          <property name="text">
           <string>*.cpp;*.h;*.ui</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="templateSearchLabel">
          <property name="text">
           <string>Шаблон поиска файлов</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="exceptionLabel">
          <property name="text">
           <string>Перечень исключений</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="exception">
          <property name="text">
           <string>moc_*.cpp;*svn*;*git*</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1137</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Меню</string>
    </property>
    <addaction name="separator"/>
    <addaction name="about"/>
    <addaction name="exit"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="palette">
    <palette>
     <active>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>147</red>
         <green>147</green>
         <blue>147</blue>
        </color>
       </brush>
      </colorrole>
     </active>
     <inactive>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </inactive>
     <disabled>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>120</red>
         <green>120</green>
         <blue>120</blue>
        </color>
       </brush>
      </colorrole>
     </disabled>
    </palette>
   </property>
   <property name="font">
    <font>
     <family>Lucida Console</family>
     <pointsize>14</pointsize>
     <italic>true</italic>
    </font>
   </property>
  </widget>
  <action name="about">
   <property name="text">
    <string>О программе</string>
   </property>
   <property name="font">
    <font>
     <family>Lucida Console</family>
     <pointsize>11</pointsize>
    </font>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="font">
    <font>
     <family>Lucida Console</family>
     <pointsize>11</pointsize>
    </font>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>templateSearch</tabstop>
  <tabstop>exception</tabstop>
  <tabstop>calculeteSpaceLine</tabstop>
  <tabstop>path</tabstop>
  <tabstop>getPath</tabstop>
  <tabstop>start</tabstop>
  <tabstop>copyBuffer</tabstop>
  <tabstop>fileCreate</tabstop>
  <tabstop>fileConfig</tabstop>
  <tabstop>tableWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>start</sender>
   <signal>clicked()</signal>
   <receiver>CodeCountLine</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>882</x>
     <y>643</y>
    </hint>
    <hint type="destinationlabel">
     <x>565</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>getPath</sender>
   <signal>clicked()</signal>
   <receiver>CodeCountLine</receiver>
   <slot>getPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1057</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>812</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exit</sender>
   <signal>triggered()</signal>
   <receiver>CodeCountLine</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>about</sender>
   <signal>triggered()</signal>
   <receiver>CodeCountLine</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>start()</slot>
  <slot>getPath()</slot>
  <slot>about()</slot>
 </slots>
</ui>
